File: prev_exams_and_ans/2017ccc/q3_exactly_electrical/ExactlyElectrical.java
import java.io.IOException;
import java.util.Scanner;

public class ExactlyElectrical {
	public static void main (String[]args) throws IOException{
		Scanner sc = new Scanner((System.in));

		int[] start = new int [2];
		int[] end = new int [2];
		
		start[0] = sc.nextInt();
		start[1] = sc.nextInt();
		
		end[0] = sc.nextInt();
		end[1] = sc.nextInt();

		int t = sc.nextInt();
		sc.close();

		int horiDiff = Math.abs(start[0] - end[0]);
		int vertDiff = Math.abs(start[1] - end[1]);

		if ( (t >= horiDiff + vertDiff) && ((horiDiff + vertDiff - t) % 2 == 0)) {
			System.out.println("Y");
		} else {
			System.out.println("N");
		}
	}
}

File: prev_exams_and_ans/2017ccc/q3_exactly_electrical/exactly_electrical.py
import sys
input = sys.stdin.readline

start = [int(x) for x in input().split()]
end = [int(x) for x in input().split()]

t = int(input())

hori_diff = abs(start[0] - end[0])
vert_diff = abs(start[1] - end[1])

if ( t >= hori_diff + vert_diff and 
    (hori_diff + vert_diff - t) % 2 == 0):
    print("Y")
else:
    print("N")